pkgs/tools/typesetting/tectonic/default.nix

```
@@ -16,20 +16,37 @@
 , openssl
 , pkg-config
 , icu
+, fetchpatch2
+, applyPatches
 }:
 
 rustPlatform.buildRustPackage rec {
   pname = "tectonic";
   version = "0.15.0";
 
-  src = fetchFromGitHub {
-    owner = "tectonic-typesetting";
-    repo = "tectonic";
-    rev = "tectonic@${version}";
-    sha256 = "sha256-xZHYiaQ8ASUwu0ieHIXcjRaH06SQoB6OR1y7Ok+FjAs=";
+  /**
+    `src` is patched up front before Cargo vendoring. We cannot reply on `patchPhase`
+    since Cargo vendoring happens before `patchPhase`. This is documented in:
+
+    https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md
+  */
```

Did you try cargoPatches?


```suggestion
  # src is patched up front before Cargo vendoring because patchPhase happens after it
```

PS: that link is super generic and you can't find anything in it. At least link a headline inside of it but IMO linking this here is unecessary.